zanata-war/src/main/java/org/zanata/service/impl/TranslationStateCacheImpl.java

```diff
@@ -181,18 +181,8 @@ public void textFlowStateUpdated(TextFlowTargetStateEvent event) {
                 new DocumentLocaleKey(event.getDocumentId(),
                         event.getLocaleId());
 
-        WordStatistic stats = documentStatisticCache.get(key);
-
-        if (stats != null) {
-            HTextFlow textFlow = getTextFlowDAO().findById(
-                    event.getTextFlowId());
-
-            stats.decrement(event.getPreviousState(),
-                    textFlow.getWordCount().intValue());
-            stats.increment(event.getNewState(),
-                    textFlow.getWordCount().intValue());
-            documentStatisticCache.put(key, stats);
-        }
+        //invalidate document statistic
+        documentStatisticCache.remove(key);
```

Comment:
We are currently doing something very efficient by altering the stats instead of re-querying everything. Is there something we can do to keep that mechanism?